### ISSUE TITLE
Docker compose profiles

### ISSUE THREAD

**Issue by iploooox** [mariotoffia/FluentDocker#289]
Hey, I was wondering if you plan to add support for using profiles?

**Comment by sdi-apmm** [comment]
Yes, please. Would be nice to choose the services you want to compose based on assigned profile

**Comment by mariotoffia** [comment]
@iploooox  good idea! -  I'll add [profiles](https://docs.docker.com/compose/profiles/) to my TODO list.

Cheers,
 Mario :)

**Comment by mariotoffia** [comment]
**Resolved in v3.0.0.** The v3 ground-up rewrite implements (or fixes) what this issue asked for.

This is a feature request (opened 2023) asking for Docker Compose profiles support, which the owner accepted onto his TODO list. v3 implements it fully and correctly, end-to-end: a public fluent builder method, a config model field, CLI driver emission of the --profile flag, and a passing unit test. I verified the actual code path rather than assuming the rewrite added it, including the critical argument ordering (--profile must be a top-level compose flag before the `up` subcommand). The produced command `docker compose -f <file> -p <proj> --profile <name> up -d` is exactly what Docker Compose expects, so the request is genuinely satisfied.

<sub>Where this lives in v3: Public API: IComposeBuilder.WithProfiles(params string[] profiles) at FluentDocker/Builders/BuilderInterfaces.cs:196-199. Builder impl: InternalComposeBuilder._profiles field (FluentDocker/Builders/InternalBuilders.cs:160), WithProfiles() (line 230), and mapping into ComposeUpConfig { ... Profiles = _profiles } in ExecuteAsync (line 254). Config model: ComposeUpConfig.Profiles (List<string>) at FluentDocker/Drivers/ComposeModels.cs:55-56. CLI emission: DockerCliComposeDriver.UpAsync iterates config.Profiles and appends `--profile {QuoteArgumentIfNeeded(profile)}` at FluentDocker/Drivers/Docker/Cli/Components/DockerCliComposeDriver.cs:35-37; ordering is correct because BuildComposeArgs (line…</sub>

Closing as part of the v3 issue sweep. If your scenario still isn't covered by v3, please reopen with a v3 reproduction and we'll take another look. Thanks for the report! 🙏
